src/clr-core/tag/tag.element.scss

```diff
@@ -105,16 +59,78 @@
   }
 }
 
-@include generateTagStyles(gray, 1);
-@include generateTagStyles(purple, 2);
-@include generateTagStyles(blue, 3);
-@include generateTagStyles(orange, 4);
-@include generateTagStyles(light-blue, 5);
+:host([color='gray']) {
+  --clr-tag-border-color: var(--clr-color-neutral-600);
 
-@include generateStatusTagStyles(info);
-@include generateStatusTagStyles(success);
-@include generateStatusTagStyles(warning);
-@include generateStatusTagStyles(danger);
+  ::slotted(cwc-badge) {
+    --clr-badge-background-color: var(--clr-color-neutral-600);
+    --clr-badge-color: var(--clr-badge-font-color-light);
```

Comment:
This just occurred to me but you _could_ instead use `--clr-color-on-neutral-600` instead of specifying `--clr-badge-font-color-light|dark`.

This would give people one place to update their colors to achieve accessible color contrast and you could get rid of all instances of `--clr-badge-font-color-light|dark`.

Reply:
I'm not sure if I follow. Are you suggesting a variable for the foreground / text color (i.e. `--clr-color-on-neutral-600` is the text color for background `--clr-color-neutral-600`? 

Doesn't that mean I'd have to create many more variables for each of the badge background colors that we are using here (and actually for all of our color variables potentially)?

Reply:
You don’t have to make any variables. They all already exist. Look at the color properties in clr-angular. 

All the colors have an accessible “on” companion property

Reply:
Wow, we should definitely be using these. Will update the PR.